### PR TITLE
[4.4] Improve error messages of uploaded package check in Joomla Update

### PR DESCRIPTION
**Pull request by richard67**
### Summary of Changes

This pull request (PR) improves the error messages from the check for the uploaded package file which was added with PR #41757 to the Joomla Update component.

1. The name of the selected update package is added to all messages. This is extremely helpful if the user has just picked the wrong file by mistake.
2. To all messages, a 2nd line is added which provides information on the right package and a link to the official Joomla download page.
3. The message on an attempt to downgrade is extended by the version numbers (package and installed).
4. The wording used in these language strings has been harmonized.

The link to the official Joomla download page points to https://downloads.joomla.org/latest , which might not be suitable for every use case, e.g. nightly builds or development versions.

But we use this link already elswhere for the same purpose, providing an entry point where people can start to look for the right package and which is suitable in most cases.

As we currently do not build the small patch packages, there are only the "Upgrade" packages available.

Later when we can build the smaller patch packages again and they will be available at the downloads pages, we might have to change the "Upgrade Package" to "Upgrade or Update Package" in the language strings touched by this PR.

#### Test 1: Upload a file which is not a ZIP file

In the Joomla Update Component, go to "Upload & Update" and then use the button to chose a file.

In the file selection dialogue, change the file type filter from "*.zip" to "All files (*.*)".
2023-09-14_upload-select-file-type-all

Select any file which is not a ZIP file, e.g. an image, a CSS file, whatever comes into your mind, and use the "Open" button to upload the file.

#### Test 2: Upload a full installation package

In the Joomla Update Component, go to "Upload & Update" and then chose a full installation package.

#### Test 3: Upload an update ZIP package for a Joomla version lower than the current Joomla version, i.e. try to downgrade

In the Joomla Update Component, go to "Upload & Update" and then chose an update ZIP package for a Joomla version lower than the current Joomla version.

#### Test 4: Upload a "hacked" update ZIP package which doesn't contain the "administrator/manifests/files/joomla.xml" file

Modify an update ZIP package with which updating should work by removing the "administrator/manifests/files/joomla.xml" file from the ZIP file, e.g. by unpacking the package, removing the XML file and packing the package again.

In the Joomla Update Component, go to "Upload & Update" and then chose the previously modified update ZIP package.

#### Test 5: Upload a "hacked" update ZIP package in which the "administrator/manifests/files/joomla.xml" file is missing the version

Modify an update ZIP package with which updating should work by removing the `<version>` element from the "administrator/manifests/files/joomla.xml" file, e.g. by unpacking the package, editing the XML file and packing the package again.

In the Joomla Update Component, go to "Upload & Update" and then chose the previously modified update ZIP package.

### Actual result BEFORE applying this Pull Request

Alert with error message without telling the name of the selected file and without a link to the official Joomla download page.

### Expected result AFTER applying this Pull Request

Alert with error message telling the name of the selected file and having a 2nd line which provides a link to the official Joomla download page and information on what to look for.

Test 3 (downgrade): The error message shows the version from the package file and the installed version used for the check.

### Link to documentations
Please select:
- [X] No documentation changes for docs.joomla.org needed

- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Review by wilsonge — APPROVED** [review]

**Comment by wilsonge** [comment]
Yay! I like this much more!

**Comment by richard67** [comment]
> Yay! I like this much more!

@wilsonge Not finished yet. Am adding the name of the selected file to the strings, too.

**Comment by richard67** [comment]
@wilsonge Now it's ready.

File: administrator/language/en-GB/com_joomlaupdate.ini [hunk 1]
```diff
@@ -177,11 +177,11 @@ COM_JOOMLAUPDATE_VIEW_UPDATE_FINALISE_HEAD_DESC="To complete the update Process
...
-COM_JOOMLAUPDATE_VIEW_UPLOAD_ERROR_INSTALL_PACKAGE="The uploaded package file looks like it is a full installation package of Joomla which can only be used for creating new sites. You can only use the \"Upgrade Package (.zip)\" to update your site."
-COM_JOOMLAUPDATE_VIEW_UPLOAD_ERROR_NO_MANIFEST_FILE="The uploaded package file is not a Joomla update package. It does not contain the \"administrator/manifests/files/joomla.xml\" file."
-COM_JOOMLAUPDATE_VIEW_UPLOAD_ERROR_NO_VERSION_FOUND="The uploaded package file is not a Joomla update package. It does not contain valid version information."
-COM_JOOMLAUPDATE_VIEW_UPLOAD_ERROR_PACKAGE_OPEN="The uploaded package file is either not a ZIP file or is corrupted."
+COM_JOOMLAUPDATE_VIEW_UPLOAD_ERROR_DOWNGRADE="The uploaded package file \"%1$s\" has a lower version \"%2$s\" than the installed version \"%3$s\". You cannot downgrade a Joomla site.<br>Check <a class='alert-link' href='https://downloads.joomla.org/latest' target='_blank' rel='noopener noreferrer'>the official Joomla download page</a> for the Joomla Upgrade Package of a <em><strong>newer version</strong></em>."
```
Comment (Quy) [inline administrator/language/en-GB/com_joomlaupdate.ini]:
```suggestion
COM_JOOMLAUPDATE_VIEW_UPLOAD_ERROR_DOWNGRADE="The uploaded package file \"%1$s\" has a lower version \"%2$s\" than the installed version \"%3$s\". You cannot downgrade a Joomla site.<br>Check <a class=\"alert-link\" href=\"https://downloads.joomla.org/latest\" target=\"_blank\" rel=\"noopener noreferrer\">the official Joomla download page</a> for the Joomla Upgrade Package of a <em><strong>newer version</strong></em>."
```
Repeat with other strings.

Comment (richard67) [reply administrator/language/en-GB/com_joomlaupdate.ini]: Done with https://github.com/joomla/joomla-cms/pull/41947/commits/66a7570a49aa7383ada7ad40a4714cf99bee53c0 .

**Review by wilsonge — APPROVED** [review]

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 66a7570a49aa7383ada7ad40a4714cf99bee53c0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41947">issues.joomla.org/tracker/joomla-cms/41947</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 66a7570a49aa7383ada7ad40a4714cf99bee53c0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41947">issues.joomla.org/tracker/joomla-cms/41947</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41947">issues.joomla.org/tracker/joomla-cms/41947</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**